### ISSUE TITLE
Add module information to `SymbolProcessorEnvironment`

### ISSUE THREAD

**Issue by dewantawsif** [google/ksp#1887]
On improving a library I maintain [FlexibleHilt](https://github.com/dewantawsif/flexible-hilt) I've decided to aggregate classes per package like below:

<div align="center">
<img src="https://github.com/google/ksp/assets/132643596/09a92614-6f29-426c-98d9-1639849915f7" width="500"/>
</div>

Now the issue is if users for whatever reason decides to have the same package in different modules it'll make duplicate interface with same name across module causing issues while building the app. Given we have access to module name we can append it to the class name to ensure it stays unique.

<div align="center">
<img src="https://github.com/google/ksp/assets/132643596/ca37bf34-ac2a-48b0-ac46-07c9bb6bbc42" width="500"/>
</div>

---
Previous issue on this are #1015, #1097 and the general answer was to pass the module name via KSP options in the gradle script.

While that is possible it adds a extra work on the end user and also another thing to keep an eye on when they rename/switch the module. As such I think it'll be better to let us handle them in the ksp processors directly. 

I would also like to note that it'll be nice if the module information had data on sub-modules too so we can get the whole path like `:core:analytics`

<div align="center">
<img src="https://github.com/google/ksp/assets/132643596/b5a6cc43-0141-4807-8e56-1842ff89b63a" width="500"/>
</div>

**Comment by undermark5** [comment]
>While that is possible it adds a extra work on the end user and also another thing to keep an eye on when they rename/switch the module

What about using `project.path` or `getProject().getPath()` when configuring it, then you don't need to worry about renames. I'm not opposed to having the information easily available in KSP, but considering it's pretty trivial to make a convention plugin that can automatically do the above, it seems that adding it directly to KSP doesn't make a lot of sense as then it simply becomes a matter of applying the convention plugin (which can also apply the KSP plugin at the same time, therefore not really adding any additional work).

**Comment by dewantawsif** [comment]
Fair point. But they did add module name recently to resolver so might as well include the project path

**Comment by jeffawx** [comment]
> and the general answer was to pass the module name via KSP options in the gradle script.

My use case is trying to generate code for both `src/main/kotlin` and `src/test/kotlin`, the generated class should have different name based on whether it's main or test e.g

- For gradle `kspKotlin` task it produce `MyClass`
- For gradle `kspTestKotlin` task it produce `MyTestClass`

Because they are in the same kotlin module, I don't know how to use ksp option to pass in current gradle configuration name. seems kspExtension and options are at project level not configuration, i.e. main, test level.
